cdhweb/pages/templatetags/core_tags.py

```diff
@@ -77,14 +82,22 @@ def _get_primary_nav_items():
     return l1_menu_items
 
 
-@register.simple_tag()
-def primary_nav_dict():
+@register.simple_tag(takes_context=True)
+def primary_nav_dict(context):
     """
     Return the primary navigation data as a dict, for use with the 'json_script' filter.
     """
     primary_nav_items = _get_primary_nav_items()
     l1_menu_item_data = [_l1_item_to_dict(item) for item in primary_nav_items]
 
+    current_path = context["request"].path
+
+    for item in l1_menu_item_data:
```

Comment:
Do we still need to do this, given we've used a different approach in the template?

Reply:
If you're asking whether we need it in both the template and the JSON blob, then answer is yes :/

Reply:
Ah, right. In my head we were using the same data for both